packages/vue/src/components/basic/SelectedFilters.jsx

```diff
@@ -67,7 +69,23 @@ const SelectedFilters = {
 			this.$emit('clear', component, value);
 		},
 		clearValues() {
-			this.clearValuesAction();
+			const { resetToDefault, componentProps, selectedValues } = this;
+			if (resetToDefault) {
+				Object.keys(selectedValues).map(component => {
+					if (
+						componentProps[component].defaultValue
+						&& !areArraysEqual(
```

Comment:
@mohdashraf010897 I would like to understand this logic. A loom video would help here.

Reply:
logic updated